tenacity/retry.py

```diff
@@ -64,14 +64,26 @@ def __call__(self, attempt):
 
 
 class retry_if_exception_type(retry_if_exception):
-    """Retries if an exception has been raised of a certain type."""
+    """Retries if an exception has been raised of one or more types."""
 
     def __init__(self, exception_types=Exception):
         self.exception_types = exception_types
         super(retry_if_exception_type, self).__init__(
             lambda e: isinstance(e, exception_types))
 
 
+class retry_until_exception_type(retry_if_exception):
+    """Retries until an exception is raised of one or more types."""
+
+    def __init__(self, exception_types=Exception):
+        self.exception_types = exception_types
+        super(retry_until_exception_type, self).__init__(
+            lambda e: not isinstance(e, exception_types))
+
+    def __call__(self, attempt):
```

Comment:
I want to retry even if no exception is raised.

I'm not sure why `retry_if_exception` needs the conditional. I would've simply removed it there, but didn't want to alter code I didn't dive deeply into. Is it an optimization?